packages/expo-ui/android/src/main/java/expo/modules/ui/CarouselView.kt

```diff
@@ -0,0 +1,231 @@
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.carousel.HorizontalMultiBrowseCarousel
+import androidx.compose.material3.carousel.HorizontalUncontainedCarousel
+import androidx.compose.material3.carousel.rememberCarouselState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.unit.dp
+import coil3.compose.AsyncImage
+import coil3.request.ImageRequest
+import coil3.request.crossfade
+import android.util.Log
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.clickable
+import androidx.compose.ui.draw.clip
+import androidx.compose.material3.MaterialTheme.typography
+import androidx.compose.material3.Text
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableIntStateOf
+import expo.modules.kotlin.AppContext
+import expo.modules.kotlin.views.ComposeProps
+import expo.modules.kotlin.views.ExpoComposeView
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
+import expo.modules.kotlin.viewevent.EventDispatcher
+import java.io.Serializable
+import androidx.core.graphics.toColorInt
+
+data class CarouselItem(
+  @Field
+  val image: String,
+  @Field
+  val title: String
+) : Record
+
+class CarouselItemPressEvent : Record, Serializable {
+  @Field
+  var index: Int = 0
+}
+
+enum class CarouselVariant(val value: String) {
+  MULTI_BROWSE("multiBrowse"),
+  UNCONTAINED("uncontained")
+}
+
+enum class TextStyle(val value: String) {
+  TITLE_LARGE("titleLarge"),
+  TITLE_MEDIUM("titleMedium"),
+  TITLE_SMALL("titleSmall"),
+  BODY_LARGE("bodyLarge"),
+  BODY_MEDIUM("bodyMedium"),
+  BODY_SMALL("bodySmall")
+}
+
+data class CarouselConfig(
+  @Field val preferredItemWidth: Float = 200f,
+  @Field val itemHeight: Float = 200f,
+  @Field val itemSpacing: Float = 8f,
+  @Field val contentPadding: Float = 16f,
+  @Field val topBottomPadding: Float = 8f,
+  @Field val cornerRadius: Float = 28f,
+  @Field val initialItemIndex: Int = 0,
+  @Field val textColor: String = "#FFFFFF",
+  @Field val textStyle: TextStyle = TextStyle.TITLE_MEDIUM
+) : Record
+
+data class CarouselViewProps(
+  val items: MutableState<List<CarouselItem>> = mutableStateOf(emptyList()),
```

Comment:
We don't need to render arbitrary RN views, but it would be nice to accept compose only children instead of getting items in props? Did you also test out that approach?